kvbc/src/reconfiguration_kvbc_handler.cpp

```diff
@@ -441,6 +443,7 @@ concord::messages::ClientStateReply KvbcClientReconfigurationHandler::buildRepli
           } else if (command_type == std::string{kvbc::keyTypes::reconfiguration_rep_main_key}) {
             concord::messages::ReplicaMainKeyUpdate cmd;
             concord::messages::deserialize(data_buf, cmd);
+            cmd.seq_num -= (cmd.seq_num < 2 * checkpointWindowSize ? 0 : 2 * checkpointWindowSize);
```

Comment:
I think the cmd seq_num should reflect the same in both, the original message and in the state reply.
The checkpoint calculation from the given sequence number can be done on the receiver side (or the original message creator should put there a checkpoint message and not a sequence number, and the filed name should be changed accordingly)

Reply:
Done